src/input.ts

```diff
@@ -9,19 +9,22 @@ export interface Input {
     crate: string;
     version: string;
     useToolCache: boolean;
-    useCache: boolean;
+    primaryKey?: string;
+    restoreKeys?: string[];
 }
 
 export function get(): Input {
     const crate = input.getInput("crate", { required: true });
     const version = input.getInput("version", { required: true });
     const useToolCache = input.getInputBool("use-tool-cache") || false;
-    const useCache = input.getInputBool("use-cache") || true;
+    const primaryKey = input.getInput("key") || undefined;
+    const restoreKeys = input.getInputAsArray("restore-keys") || undefined;
```

Comment:
I don't think it is a good idea to force users to provide all necessary cache keys, as Action can do that by itself; grab the crate name, version and current runner name (ex. `ubuntu-18.04`), join them together - we got a cache key.
That also mean that we will need to use `resolveVersion` if there is no version specifically set in the Action arguments or set to `"latest"`, for example.

Ideally, from the user side I would expect to write
```yaml
- uses: actions-rs/install@v0.2
  with:
    crate: cargo-make
```

and it will handle all the stuff by itself.

Reply:
My thought behind adding those keys was that there is currently no way to force-update a cache. If a new cache should be created, the cache has to get a new name. With `actions/cache` in rust projects, this is for example done by hashing the `Cargo.lock` files.

But when caching binaries I guess only naming the hash with the version number should do the trick. There should be no need to add a hash to the cache name.

Should I remove the keys completely or should I just make them optional?